tap_mysql/__init__.py

```diff
@@ -408,13 +409,19 @@ def sync_table(connection, db, table, columns, state):
         yield state.make_state_message()
 
 
-
 def generate_messages(con, raw_selections, raw_state):
     indexed_schema = index_schema(discover_schemas(con))
     state = State(raw_state, raw_selections)
-    for stream in raw_selections['streams']:
+
+    streams = raw_selections['streams']
+    if state.current_stream:
+        streams = dropwhile(lambda s: s['stream'] != state.current_stream, streams)
```

Comment:
if this `dropwhile` doesn't take `s['selected']` into account, then the effect of using `current_stream` is degraded. this is due to the fact that the `raw_selections` stream list may contain duplicate stream names (though we make the assumption that the list of _selected_ stream names contains no dupes).

e.g.

```
state['current_stream'] == 'foo'

# raw_selections
[
  ...
  {database_name: 'a', stream: 'foo'},                   <-- we think this is the current_stream, so we pick up here
  ...                                                    <-- a bunch of streams we replicated on the last run
  {database_name: 'b', stream: 'foo', selected: 'true'}, <-- the 'real' current_stream, where we should have picked up from
  ...
]
```